app/controllers/concerns/foreman/controller/foreman_url_renderable.rb

```diff
@@ -0,0 +1,76 @@
+module Foreman::Controller::ForemanUrlRenderable
+  class ForemanUrlProvider
+    attr_accessor :provider
+
+    def initialize(provider = nil)
+      @provider = nil
+    end
+
+    def override?(host)
+      @provider.override?(host)
+    end
+
+    def foreman_url(*args)
+      if @provider.present?
+        @provider.foreman_url(*args)
+      else
+        raise 'provider missing'
```

Comment:
This must be ForemanException, properly formatted. Also expand on which provider are we erroring out about.

http://projects.theforeman.org/projects/foreman/wiki/Translating#Exceptions